gym_microrts/petting_zoo_api.py

```diff
@@ -0,0 +1,162 @@
+import functools
+import random
+from copy import deepcopy
+from pdb import set_trace
+
+import gym
+import numpy as np
+from pettingzoo import AECEnv
+from pettingzoo.test import api_test
+from pettingzoo.utils import agent_selector
+
+from gym_microrts import microrts_ai
+from gym_microrts.envs.vec_env import MicroRTSGridModeSharedMemVecEnv
+
+
+class PettingZooMicroRTSGridModeSharedMemVecEnv(AECEnv, MicroRTSGridModeSharedMemVecEnv):
```

Comment:
Could you also implement the `action_mask` interface? See https://www.pettingzoo.ml/classic/checkers as an example.

Reply:
Sure!

Reply:
It's already implemented here so you can just apply a wrapper to it.

https://github.com/vwxyzjn/gym-microrts/blob/3d7a42f46efbd39a0b806388b8a445fbee48d00f/gym_microrts/envs/vec_env.py#L217-L221

Reply:
Do you think creating a wrapper function is sufficient, or should it be directly applied in the step function and not be exposed to the user?